Spec/RealtimeClientConnection.swift

```diff
@@ -2649,6 +2649,118 @@ class RealtimeClientConnection: QuickSpec {
+                                        client.connection.once(.connected) { _ in
+                                            expect(client.connection.id).toNot(equal(connectionId))
+                                            done()
+                                        }
+                                    }
+                                }
+                                client.onDisconnected()
+                            }
+                        }
+                    }
+                    // RTN15g3
+                    it("reattaches to the same channels after a new connection has been established") {
+                        client = AblyTests.newRealtime(options)
+                        client.connect()
+                        defer { client.close() }
+                        let channelName = "test-reattach-after-ttl"
+                        let channel = client.channels.get(channelName)
+                        
+                        waitUntil(timeout: testTimeout) { done in
+                            client.connection.once(.connected) { _ in
+                                connectionId = client.connection.id!
+                                client.connectionStateTtl = customTtlInterval
+                                client.maxIdleInterval = customIdleInterval
+                                channel.attach { error in
+                                    if let error = error {
+                                        fail(error.message)
+                                    }
+                                    expect(channel.state).to(equal(ARTRealtimeChannelState.attached))
+                                }
+                                client.connection.once(.disconnected) { _ in
+                                    client.connection.once(.connecting) { _ in
+                                        client.connection.once(.connected) { _ in
+                                            expect(client.connection.id).toNot(equal(connectionId))
+                                            channel.once(.attached) { _ in
+                                                done()
+                                            }
+                                        }
+                                    }
+                                }
+                                client.onDisconnected()
+                            }
+                        }
+                    }
+                }
+                
+                // RTN15g2
+                context("when connection (ttl + idle interval) period has NOT passed since last activity") {
+                    let options = AblyTests.commonAppSetup()
+                    options.disconnectedRetryTimeout = 5.0
+                    var client: ARTRealtime!
+                    var connectionId = ""
+                    let customTtlInterval: TimeInterval = 20.0
```

Comment:
Why even set this, surely a standard connection should reconnect automatically without any custom settings?

Reply:
@mattheworiordan good spot. This is not needed.